File: lab3/src/main/java/lab3/myquest/locations/Unconscious.java
package lab3.myquest.locations;

import lab3.functionality.Location;
import lab3.functionality.Player;

public class Unconscious extends Location {
    public Unconscious(String name) {
        super(name, p -> p.getName() + " пытается заглянуть внутрь себя");
    }

    @Override
    public String enter(Player player) {
        return String.format("Удар был настолько силен, что %s ошалел от боли и на некоторое время перестал трепыхаться в воздухе.", player.getName());
    }
}

File: lab4/src/main/kotlin/lab4/myQuest/locations/Unconscious.kt
package lab4.myQuest.locations

import lab4.functionality.ActionFactory
import lab4.functionality.Location
import lab4.functionality.Player

class Unconscious : Location("Пустота?", { "${it.name} пытается заглянуть внутрь себя." }) {
    override fun enter(player: Player): String {
        player.damage()
        player.actions.add(
            ActionFactory().changeLocation(
                "Прийти в себя",
                { "Придя постепенно в себя, ${it.name} убедился, что висит в какой-то нелепой позе посреди комнаты, между полом и потолком." },
                Rocket(),
                true
            )
        )
        return ""
    }

}